compiler/src/main/scala/org/scalajs/nscplugin/GenJSCode.scala

```diff
@@ -807,14 +807,27 @@ abstract class GenJSCode[G <: Global with Singleton](val global: G)
       generatedClasses += newClassDef
 
       // Construct inline class definition
-      val js.JSMethodDef(_, _, ctorParams, ctorBody) =
-        constructor.getOrElse(throw new AssertionError("No ctor found"))
 
-      val jsSuperClassParam = js.ParamDef(freshLocalIdent("super")(pos),
-          NoOriginalName, jstpe.AnyType, mutable = false, rest = false)(pos)
-      def jsSuperClassRef(implicit pos: ir.Position) =
-        jsSuperClassParam.ref
+      val jsClassCaptures = origJsClass.jsClassCaptures.getOrElse {
+        throw new AssertionError(
+            s"no class captures for anonymous JS class at $pos")
+      }
+      val js.JSMethodDef(_, _, ctorParams, ctorBody) = constructor.getOrElse {
+        throw new AssertionError("No ctor found")
+      }
+      assert(ctorParams.isEmpty,
+          s"non-empty constructor params for anonymous JS class at $pos")
+
+      /* The first class capture is always a reference to the super class.
+       * This is enforced by genJSClassCapturesAndConstructor.
+       */
+      def jsSuperClassRef(implicit pos: ir.Position): js.VarRef =
+        jsClassCaptures.head.ref
 
+      /* The `this` reference.
+       * FIXME This could clash with a local variable of the constructor or a JS
+       * class capture. How do we avoid this?
```

Comment:
This was already a problem before this PR. The new comment is just because I realized the problem now.